pkg/config/config.go

```diff
@@ -206,57 +204,56 @@ func (c *Config) updateComputedValues() error {
 }
 
 func (c *Config) validate() error {
-	if len(c.ApiServer.SubjectAltNames) > 0 {
-		// Any entry in SubjectAltNames will be included in the external access certificates.
-		// Any of the hostnames and IPs (except the node IP) listed below conflicts with
-		// other certificates, such as the service network and localhost access.
-		// The node IP is a bit special. Apiserver k8s service, which holds a service IP
-		// gets resolved to the node IP. If we include the node IP in the SAN then we have
-		// an ambiguity, the same IP matches two different certificates and there are errors
-		// when trying to reach apiserver from within the cluster using the service IP.
-		// Apiserver will decide which certificate to return to client hello based on SNI
-		// (which client-go does not use) or raw IP mappings. As soon as there is a match for
-		// the node IP it returns that certificate, which is the external access one. This
-		// breaks all pods trying to reach apiserver, as hostnames dont match and the certificate
-		// is invalid.
-		u, err := url.Parse(c.ApiServer.URL)
-		if err != nil {
-			return fmt.Errorf("failed to parse cluster URL: %v", err)
-		}
-		if u.Hostname() == "localhost" || u.Hostname() == "127.0.0.1" {
-			if stringSliceContains(c.ApiServer.SubjectAltNames, "localhost", "127.0.0.1") {
-				return fmt.Errorf("subjectAltNames must not contain localhost, 127.0.0.1")
-			}
-		} else {
-			if stringSliceContains(c.ApiServer.SubjectAltNames, c.Node.NodeIP) {
-				return fmt.Errorf("subjectAltNames must not contain node IP")
-			}
-			if !stringSliceContains(c.ApiServer.SubjectAltNames, u.Host) || u.Host != c.Node.HostnameOverride {
-				return fmt.Errorf("Cluster URL host %v is not included in subjectAltNames or nodeName", u.String())
-			}
-		}
-		if stringSliceContains(
-			c.ApiServer.SubjectAltNames,
-			"kubernetes",
-			"kubernetes.default",
-			"kubernetes.default.svc",
-			"kubernetes.default.svc.cluster.local",
-			"openshift",
-			"openshift.default",
-			"openshift.default.svc",
-			"openshift.default.svc.cluster.local",
-			c.ApiServer.AdvertiseAddress,
-		) {
-			return fmt.Errorf("subjectAltNames must not contain apiserver kubernetes service names or IPs")
-		}
-	}
-
 	if c.Etcd.MemoryLimitMB > 0 && c.Etcd.MemoryLimitMB < EtcdMinimumMemoryLimit {
 		return fmt.Errorf("etcd.memoryLimitMB value %d is below the minimum allowed %d",
 			c.Etcd.MemoryLimitMB, EtcdMinimumMemoryLimit,
 		)
 	}
-
+	if len(c.ApiServer.SubjectAltNames) == 0 {
```

Comment:
Flipping this condition around will cause problems when more validations get added later.

Reply:
Fair point. I've reverted the change and added a linter skip tag.

Reply:
I don't see a linter skip tag here. Was a change lost in a rebase?